### PR TITLE
Boost: Add a gadget for measuring the performance improvement Super Cache brings

### PR DESCRIPTION
**Pull request by thingalon**
This PR adds a new tool to Jetpack Boost, allowing users to measure the speed of Super Cache if they have it installed alongside Boost.

It also includes a prompt for users to enable Super Cache if it's installed, but not enabled.

The styles are based on the designs in p1HpG7-hq3-p2, with a mind to build out the rest of that flow once we're further along.

Here is a video of the speed gadget (and cache enabling prompt) in action:

https://user-images.githubusercontent.com/1369626/196329311-bbcec36f-7fd8-461b-bda8-c6582e3a971a.mov

Notes:
- The "Super Cache is not enabled" prompt is dismissable.
- The test doesn't automatically run, so that we don't generate traffic on the user's site for it when visiting the dashboard.
- The tool is placed in a location where we can put information about detected caches, and a prompt to install Super Cache in the future if users have no caching solution. 
- The `Notice` is implemented as a new storybook component!
- This is my first Storybook component, so please be kind ;)

#### Changes proposed in this Pull Request:
- Adds a new storybook component `Notice` based on the designs here: p1HpG7-hq3-p2
- Uses the new `Notice` component to inform users if Super Cache needs to be enabled.
- Offers users a new way to test the speed of their cache.

#### Other information:
- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
p1HpG7-hq3-p2

#### Does this pull request change what data or activity we track or use?
Adds tracks events for running Super Cache tests, so that we can measure popularity of the feature, and general effectiveness of Super Cache.

#### Testing instructions:
* Install Super Cache, and enable the plugin - but don't turn on Caching. 
* Visit the Boost dashboard, and verify there is a warning there prompting users to enable their cache. 
* Verify the Set up button sends users to the Super Cache settings page
* Verify the notice is dismissable, and stays dismissed.
* Turn on Caching
* Go to the Boost dashboard
* Verify that you can measure how much time Super Cache is saving your users. :)

### CODE REVIEW

**Comment by coder-karen** [comment]
This doesn't seem to touch the Jetpack plugin (or WP Super Cache), does it need review from Crew?

**Comment by thingalon** [comment]
> This doesn't seem to touch the Jetpack plugin (or WP Super Cache), does it need review from Crew?

You're right, it doesn't. I've updated the tags.

**Comment by oskosk** [comment]
I was testing this an realized that super cache was installed and active on my site but the widget was not showing. Then realized `wp_cache_is_enabled()` was returning `false` and remembered I had it disabled here

image

Should we show something about their integration, prompting users to enable cache for getting better score or something like that ?

**Comment by oskosk** [comment]
Pushed rebase after merging #26715. I'm updating now the screenshot on this PR to show how it looks right now

**Comment by thingalon** [comment]
@oskosk - who is the best person / team to ping for reviews when there are new components in a build?
@Automattic/heart-of-gold - can I please get a team review on this

File: projects/plugins/boost/app/assets/src/js/stores/config.ts
Comment (haqadn) [inline projects/plugins/boost/app/assets/src/js/stores/config.ts]: It would give a better perception of speed if we first updated the state and then make the ajax call.

```suggestion
	dismiss: async ( name: string ) => {
		dismissedPopOutStore.update( dismissals => [ ...dismissals, name ] );

		await makeAdminAjaxRequest( {
			action: 'set_show_score_prompt',
			id: name,
			value: 'false',
			// eslint-disable-next-line camelcase
			nonce: Jetpack_Boost.showScorePromptNonce,
		} );
```

**Review by haqadn — DISMISSED** [review]
I think it is good to go with the minor improvement.

File: projects/plugins/boost/app/assets/src/js/stores/config.ts
Comment (thingalon) [reply projects/plugins/boost/app/assets/src/js/stores/config.ts]: Sure, I'm ok with that. Thanks!

**Review by pyronaur — COMMENTED** [review]
Just noting that there's a test timing out. Otherwise LGTM.

**Review by oskosk — DISMISSED** [review]
LGTM!

**Review by oskosk — APPROVED** [review]

**Pull request merged**
